### PR TITLE
Gea 1257: improve audit scripts

### PR DESCRIPTION
**Pull request by dbenders**
Complete refactor of the three scripts: `dump_audit`, `verify_audit` and `deep_verify`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,34 +91,82 @@ else:
...
+usage: python -m pangea.verify_audit [-h] [--file PATH]
+
+Pangea Audit Verifier
+
+options:
+  -h, --help            show this help message and exit
+  --file PATH, -f PATH  Input file (default: standard input).
+```
+
+It accept multiple file formats:
```
Comment (govind-salinas-pangea) [inline README.md:106]: accepts

File: README.md [hunk 2]
```diff
@@ -91,34 +91,82 @@ else:
 
 #### Verify audit data
 
-You can provide a single event (obtained from the PUC) or the result from a search call.
-In the latter case, all the events are verified.
+Verify that an event or a list of events has not been tampered. Usage:
```
Comment (govind-salinas-pangea) [inline README.md:94]: tampered with

File: README.md [hunk 3]
```diff
@@ -91,34 +91,82 @@ else:
...
+  --token TOKEN, -t TOKEN
+                        Pangea token (default: env PANGEA_TOKEN)
+  --base-domain BASE_DOMAIN, -d BASE_DOMAIN
+                        Pangea base domain (default: env PANGEA_DOMAIN)
+  --config-id CONFIG_ID, -c CONFIG_ID
+                        Audit config id (default: env PANGEA_AUDIT_CONFIG_ID)
+  --file FILE, -f FILE  Event input file. Must be a collection of JSON Objects separated by newlines
 ```
 
+It accept multiple file formats:
```
Comment (govind-salinas-pangea) [inline README.md:164]: accepts

File: pangea/deep_verify.py [hunk 4]
```diff
@@ -123,8 +103,8 @@ def verify_hash(data: dict, data_hash: str) -> bool:
     succeeded = False
     try:
         data_canon = audit_util.canonicalize_json(data)
-        computed_hash = audit_util.hash_data(data_canon)
-        computed_hash_dec = audit_util.decode_hash(computed_hash)
+        computed_hash_dec = audit_util.hash_bytes(data_canon)
+        # computed_hash_dec = audit_util.decode_hash(computed_hash)
```
Comment (govind-salinas-pangea) [inline pangea/deep_verify.py:107]: if this line is not needed, remove it.

File: pangea/deep_verify.py [hunk 5]
```diff
@@ -146,90 +131,166 @@ def verify_membership_proof(node_hash: str, root_hash: str, proof: str) -> bool:
...
+
+        cold_indexes.add(leaf_index)
+
+        cold_path_size: t.Optional[int] = None
+        hot_out_of_order: set[int] = set()
+        for i, event in enumerate(events_by_idx):
+            cnt += 1
+            tree_size = get_tree_size(event)
+            if tree_size not in root_hashes:
+                root_hashes[tree_size] = get_root_hash(audit, tree_size)
```
Comment (govind-salinas-pangea) [inline pangea/deep_verify.py:185]: could this fail, say if tree_size was the wrong size?

File: pangea/verify_audit.py [hunk 6]
```diff
@@ -77,14 +77,14 @@ def log_section(msg: str):
...
+def _verify_hash(data: dict, data_hash: str) -> t.Optional[bool]:
     log_section("Checking data hash")
     try:
         logger.debug("Canonicalizing data")
         data_canon = canonicalize_json(data)
         logger.debug("Calculating hash")
-        computed_hash = hash_data(data_canon)
-        computed_hash_dec = decode_hash(computed_hash)
+        computed_hash_dec = hash_bytes(data_canon)
+        # computed_hash_dec = decode_hash(computed_hash)
```
Comment (govind-salinas-pangea) [inline pangea/verify_audit.py:87]: remove line if its not needed.

File: pangea/deep_verify.py (hunk 5 above)
Comment (dbenders) [reply pangea/deep_verify.py:185]: yes, the script panics in that case. I think it is a pretty bad error, I shouldn't continue

**Comment by dbenders** [comment]
@govind-salinas-pangea @govind-salinas-pangea this is ready for merge, please approve and merge

**Review by govind-salinas-pangea — APPROVED** [review]

**Pull request merged**